libs/recipes/sodium/rules.mk

```
@@ -0,0 +1,21 @@
+LIBSODIUM_VERSION = 1.0.18
+LIBSODIUM_TARBALL = $(DOWNLOAD)/libsodium-$(LIBSODIUM_VERSION).tar.gz
+LIBSODIUM_URL = https://download.libsodium.org/libsodium/releases/libsodium-${LIBSODIUM_VERSION}.tar.gz
+
+.PHONY: libsodium
+libsodium: $(LIBSODIUM_WASM_LIB)
+
+$(LIBSODIUM_TARBALL):
+	mkdir -p $(DOWNLOAD)
+	wget $(LIBSODIUM_URL) -O $@
+
+$(LIBSODIUM_WASM_LIB): $(LIBSODIUM_TARBALL) 
+	mkdir -p $(BUILD)/libsodium-$(LIBSODIUM_VERSION)/build
+	tar -C $(BUILD) -xf $(LIBSODIUM_TARBALL)
+	cd $(BUILD)/libsodium-$(LIBSODIUM_VERSION)/build && \
+	  emconfigure ../configure \
+	    --enable-shared=no \
+	    --enable-static=yes \
```

```suggestion
	    --enable-static=yes \
	    --without-pthreads \
```

With Emscripten, support for both `-pthread` and `-s SIDE_MODULE=1` at the same time is currently experimental
and does not work with R packages for webR. Without this tweak, the `sodium` package fails to load.

---

Good to know, thanks